src/OpenTelemetry.Contrib.Extensions.AWSXRay/Resources/AWSEBSResourceDetector.cs

```diff
@@ -0,0 +1,83 @@
+﻿// <copyright file="AWSEBSResourceDetector.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+#if NETSTANDARD
+using System.Runtime.InteropServices;
+#endif
+using OpenTelemetry.Contrib.Extensions.AWSXRay.Resources.Models;
+
+namespace OpenTelemetry.Contrib.Extensions.AWSXRay.Resources
+{
+    /// <summary>
+    /// Resource detector for application running in AWS ElasticBeanstalk environment.
+    /// </summary>
+    public class AWSEBSResourceDetector : IResourceDetector
+    {
+        private const string AWSEBSMetadataWindowsFilePath = "C:\\Program Files\\Amazon\\XRay\\environment.conf";
+        private const string AWSEBSMetadataLinuxFilePath = "/var/elasticbeanstalk/xray/environment.conf";
+
+        /// <summary>
+        /// Detector the required and optional resource attributes from AWS ElasticBeanstalk.
+        /// </summary>
+        /// <returns>List of key-value pairs of resource attributes.</returns>
+        public IEnumerable<KeyValuePair<string, object>> Detect()
+        {
+            List<KeyValuePair<string, object>> resourceAttributes = null;
+
+            try
+            {
+                string filePath = null;
+#if NETSTANDARD
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                {
+                    filePath = AWSEBSMetadataWindowsFilePath;
+                }
+                else
+                {
+                    filePath = AWSEBSMetadataLinuxFilePath;
+                }
+#else
+                filePath = AWSEBSMetadataWindowsFilePath;
+#endif
+
+                var metadata = this.GetEBSMetadata(filePath);
```

Comment:
I think we can unit test this by using a mock file system. For instance, if we use the interfaces defined in https://www.nuget.org/packages/System.IO.Abstractions/13.2.47 to interact with the file system, maybe we can add a new utility method where the file system is configurable, and we can pass in a fake file system to mock the responses it gives?

We can also extract all of this logic into a helper method if we need to pass in a new file system for testing, since I know we can't touch the `Detect()` API.